tests/conftest.py

```diff
@@ -429,6 +429,112 @@ def finalizer():
     return factory
 
 
+@pytest.fixture()
+def service_account_factory(request):
+    """
+    Create a service account
+    """
+    instances = []
+    active_service_account_obj = None
+
+    def factory(project=None):
+        """
+        Args:
+            project (object): ocs_ci.ocs.resources.ocs.OCS instance
+                of 'Project' kind.
+
+        Returns:
+            object: helpers.create_serviceaccount instance.
+        """
+        nonlocal active_service_account_obj
+        if active_service_account_obj:
+            return active_service_account_obj
+        else:
+            sa_obj = helpers.create_serviceaccount(
+                namespace=project.namespace,
+            )
+            sa_obj.project = project
+            active_service_account_obj = sa_obj
+            helpers.add_scc_policy(sa_name=sa_obj.name, namespace=project.namespace)
+            assert sa_obj, "Failed to create serviceaccount"
+            instances.append(sa_obj)
+            return sa_obj
+
+    def finalizer():
+        """
+        Delete the service account
+        """
+        for instance in instances:
+            helpers.remove_scc_policy(
+                sa_name=instance.name,
+                namespace=instance.namespace
+            )
+            instance.delete()
+            instance.ocp.wait_for_delete(resource_name=instance.name)
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def dc_pod_factory(
```

Comment:
Would it make sense to enhance `pod_factory` with new parameters like `service_account` and `dc_deployment` instead of creating this new fixture?

Reply:
personally, I feel we should have separate dc_pod_factory it will be easy for everyone to use if we suppose to make a change in pod_factory users  need to add additional data to pvc_factory for setting up dc_pods